### PR TITLE
Feat/mc dropout

### PR DESCRIPTION
**Pull request by hrzn**
This is a new take at https://github.com/unit8co/darts/pull/714

### CODE REVIEW

File: docs/userguide/forecasting_overview.md [hunk 1]
```diff
@@ -196,6 +197,40 @@ pred.plot(label='forecast')
 
 ![TCN quantile regression](./images/probabilistic/example_tcn_quantile.png)
 
+### Capturing model uncertainty using Monte Carlo Dropout
+In Darts, dropout can also be used an additional way to capture model uncertainty, following the approach described in [this paper](https://arxiv.org/abs/1506.02142). This is sometimes referred to as *epistemic uncertainty*, and can be seen as a way to marginalize over a family of models represented by all the different dropout activation functions.
```
Comment (gdevos010) [inline docs/userguide/forecasting_overview.md:201]: I recommend writing out the paper's title. It can help with Google searches.

**Comment by gdevos010** [comment]
Is there any conflict between using MC and a likelihood?

File: darts/utils/torch.py [hunk 2]
```diff
@@ -20,6 +23,45 @@
...
+    from the space of all available models. This provides mathematical
+    grounds to reason about the model’s uncertainty and, as it turns out,
+    often improves its performance.
+    """
+
+    # We need to init it to False as some models may start by
+    # a validation round, in which case MC dropout is disabled.
+    mc_dropout_enabled: bool = False
+
+    def train(self, mode: bool = True):
```
Comment (gdevos010) [inline darts/utils/torch.py:44]: I suggest renaming this to `train`. `mode` is not descriptive as a bool

File: docs/userguide/forecasting_overview.md (hunk 1 above)
Comment (hrzn) [reply docs/userguide/forecasting_overview.md:201]: Good idea 👍

File: darts/utils/torch.py (hunk 2 above)
Comment (hrzn) [reply darts/utils/torch.py:44]: I agree, unfortunately this is coming from PyTorch and we have to follow the nomenclature here, because we are inheriting `nn.Dropout` and this function might be called from PyTorch using `mode` as an argument name.

File: docs/userguide/forecasting_overview.md [hunk 3]
```diff
@@ -196,6 +197,40 @@ pred.plot(label='forecast')
 
 ![TCN quantile regression](./images/probabilistic/example_tcn_quantile.png)
 
+### Capturing model uncertainty using Monte Carlo Dropout
+In Darts, dropout can also be used an additional way to capture model uncertainty, following the approach described in [1]. This is sometimes referred to as *epistemic uncertainty*, and can be seen as a way to marginalize over a family of models represented by all the different dropout activation functions.
```
Comment (gnwhr) [inline docs/userguide/forecasting_overview.md:201]: small typo: *as* an additional way
